<launch>
<!-- todo: clean args -->

  <!-- args corresponding to environment variables -->
  <arg name="RUN_TYPE" default="$(optenv RUN_TYPE)" />
  <arg name="PROFILER" default="$(optenv PROFILER false)" />
  <arg name="UAV_NAME" default="$(optenv UAV_NAME)" />
  <arg name="UAV_MASS" default="$(optenv UAV_MASS)" />
  <arg name="BODY_DISTURBANCE_X" default="$(optenv INITIAL_DISTURBANCE_X 0.0)" />
  <arg name="BODY_DISTURBANCE_Y" default="$(optenv INITIAL_DISTURBANCE_Y 0.0)" />
  <arg name="LOGGER_DEBUG" default="$(optenv LOGGER_DEBUG false)" />
  <arg name="OLD_PX4_FW" default="$(optenv OLD_PX4_FW false)" />

    <!-- other args -->
  <arg name="debug" default="false" />
  <arg name="profile" default="true" />
  <arg name="body_frame" default="fcu" />
  <arg name="standalone" default="true" />
  <arg name="bond" default="$(optenv BOND true)" />
  <arg name="nodelet_manager_name" default="$(optenv NODELET_MANAGER)" />

  <arg name="custom_config" default="" />
  <arg name="world_config" default="" />
  <arg name="platform_config" default="" />
  <arg name="network_config" default="" />

  <env name="ROSCONSOLE_CONFIG_FILE" if="$(eval arg('LOGGER_DEBUG'))" value="$(find mrs_uav_managers)/config/debug_verbosity.yaml" />

  <arg name="g" default="9.81" />

  <arg     if="$(arg debug)" name="launch_prefix" value="debug_roslaunch" />
  <arg unless="$(arg debug)" name="launch_prefix" value="" />

  <arg     if="$(arg bond)" name="bond_suffix" value="" />
  <arg unless="$(arg bond)" name="bond_suffix" value="--no-bond" />

  <arg     if="$(eval arg('standalone') or arg('debug'))" name="nodelet" value="standalone" />
  <arg unless="$(eval arg('standalone') or arg('debug'))" name="nodelet" value="load" />
  <arg     if="$(eval arg('standalone') or arg('debug'))" name="nodelet_manager" value="" />
  <arg unless="$(eval arg('standalone') or arg('debug'))" name="nodelet_manager" value="$(arg nodelet_manager_name)" />


  <group ns="$(arg UAV_NAME)">
    <node pkg="nodelet" type="nodelet" name="safety_area_manager" args="$(arg nodelet) mrs_uav_managers/SafetyAreaManager $(arg nodelet_manager) $(arg bond_suffix)" output="screen" launch-prefix="$(arg launch_prefix)">


      <param if="$(eval arg('world_config') == '')" name="world_config" value="" />
      <param if="$(eval not arg('world_config') == '' and arg('world_config')[0] == '/')" name="world_config" value="$(arg world_config)" />
      <param if="$(eval not arg('world_config') == '' and not arg('world_config')[0] == '/')" name="world_config" value="$(env PWD)/$(arg world_config)" />
      

    <!-- Subscribers -->
    <remap from="~hw_api_capabilities_in" to="hw_api/capabilities" />

    <remap from="chatter" to="chatter_remapped" />

    <!-- todo: add params, remappings -->
    </node>
  </group>
</launch>